modules/Microsoft.Compute/virtualMachines/deploy.bicep

```diff
@@ -166,7 +166,7 @@ param enableServerSideEncryption bool = false
 @description('Optional. Specifies whether extension operations should be allowed on the virtual machine. This may only be set to False when no extensions are present on the virtual machine.')
 param allowExtensionOperations bool = true
 
-@description('Optional. Required if domainName is specified. Password of the user specified in domainJoinUser parameter.')
+@description('Optional. Required if domainName is specified. Password of the user specified in user parameter.')
```

Comment:
Thanks for fixing it @jriekse5555 . Also this same issue exists on the vm scale sets module. Can you update it there as well please? If you can also generate the readme again from the [script](https://github.com/Azure/ResourceModules/blob/main/utilities/tools/Set-ModuleReadMe.ps1) and would be great if you can attach a successful test pipeline status badge for both modules after the change.

Reply:
@ahmadabdalla Changes made in vm scale sets readme and module. Also tweaked the parameter descriptions in all four files from domainName to name. 

I haven't used the referenced script previously so would need more detailed instructions on how to leverage it.

Reply:
@jriekse5555 no problems:
If you load the script into your PowerShell session. Then try performing the below:

```pwsh
Set-ModuleReadMe -TemplateFilePath 'Path to your module directory\deploy.bicep'
```

So you need to run this script across both the VM and VMSS module, and it should replace the existing readme for you. Also just make sure to test the modules as well

Reply:
There appears to be a bit more to running that powershell script from the cloned copy I made of the PR fork (see powershell output below). With the parameter descriptions in the .bicep matching the readme (for both VM and VMSS) in theory it shouldn't change the readme file. 

Join-Path:
Line |
  34 |      . (Join-Path $PSScriptRoot 'helper/Merge-FileWithNewContent.ps1')
     |                   ~~~~~~~~~~~~~
     | Cannot bind argument to parameter 'Path' because it is an empty string.
Split-Path:
Line |
  35 |      . (Join-Path (Split-Path $PSScriptRoot -Parent) 'pipelines' 'shar …
     |                               ~~~~~~~~~~~~~
     | Cannot bind argument to parameter 'Path' because it is an empty string.
WARNING: A new Bicep release is available: v0.8.9. Upgrade now by running "az bicep upgrade".
Get-NestedResourceList:
Line |
  25 |      $RelevantResourceTypeObjects = Get-NestedResourceList $TemplateFi …
     |                                     ~~~~~~~~~~~~~~~~~~~~~~
     | The term 'Get-NestedResourceList' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
Merge-FileWithNewContent:
Line |
  56 |          $updatedFileContent = Merge-FileWithNewContent -oldContent $R …
     |                                ~~~~~~~~~~~~~~~~~~~~~~~~
     | The term 'Merge-FileWithNewContent' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
Set-ParametersSection:
Line |
 118 |          $readMeFileContent = Set-ParametersSection @inputObject
     |                                                     ~~~~~~~~~~~~
     | Cannot bind argument to parameter 'ReadMeFileContent' because it is null.
Set-OutputsSection:
Line |
 128 |          $readMeFileContent = Set-OutputsSection @inputObject
     |                                                  ~~~~~~~~~~~~
     | Cannot bind argument to parameter 'ReadMeFileContent' because it is null.
Set-DeploymentExamplesSection:
Line |
 140 |  …       $readMeFileContent = Set-DeploymentExamplesSection @inputObject
     |                                                             ~~~~~~~~~~~~
     | Cannot bind argument to parameter 'ReadMeFileContent' because it is null.
Set-TableOfContent:
Line |
 149 |          $readMeFileContent = Set-TableOfContent @inputObject
     |                                                  ~~~~~~~~~~~~
     | Cannot bind argument to parameter 'ReadMeFileContent' because it is null.

Reply:
@jriekse5555 will test this out and let you know

Reply:
Hey @jriekse5555, Which version of az cli do you have installed? To check
`az version`
If 2.37 you may have the issue tracked in https://github.com/Azure/ResourceModules/issues/1471. In that case try to upgrade to 2.38 with
`az upgrade`

Then rerun the script and let us know!

Reply:
@jriekse5555 I've replayed your changes locally and we do not need to run the readme script as it produced the same hash. However testing the module through pester is going to break when merged as we need to uplift the network interface module api version. See below

![image](https://user-images.githubusercontent.com/28486158/181444074-90348f80-ebed-4664-92d0-8617710242d1.png)

We need to fix that first before merging this change. I will raise an issue for that and then come back to this one.

Reply:
@jriekse5555 PR raised for #1722 to unblock this PR.

Reply:
PR Unlocked.